src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp

```diff
@@ -1230,9 +1230,9 @@ namespace OpenMS
 
     // print some adduct stats:
     OPENMS_LOG_INFO << "Hits by adduct: #peaks explained (# matching db entries)'\n";
-    for (std::map<String, UInt>::const_iterator it = adduct_stats.begin(); it != adduct_stats.end(); ++it)
+    for (const auto& [name, num_entries] : adduct_stats)
```

Comment:
This structured binding doesn't compile on MacOS:

/Users/builder/jenkins/ws/openms/PR/bldtst/87227eb0/source/src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp:1235:35: error: reference to local binding 'name' declared in enclosing function 'OpenMS::AccurateMassSearchEngine::exportMzTab_'
      OPENMS_LOG_INFO << "  '" << name << "' : " << adduct_stats_unique[name].size() << " (" << num_entries << ")\n";
                                  ^
/Users/builder/jenkins/ws/openms/PR/bldtst/87227eb0/source/src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp:1233:23: note: 'name' declared here
    for (const auto& [name, num_entries] : adduct_stats)
                      ^
/Users/builder/jenkins/ws/openms/PR/bldtst/87227eb0/source/src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp:1235:73: error: reference to local binding 'name' declared in enclosing function 'OpenMS::AccurateMassSearchEngine::exportMzTab_'
      OPENMS_LOG_INFO << "  '" << name << "' : " << adduct_stats_unique[name].size() << " (" << num_entries << ")\n";
                                                                        ^
/Users/builder/jenkins/ws/openms/PR/bldtst/87227eb0/source/src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp:1233:23: note: 'name' declared here
    for (const auto& [name, num_entries] : adduct_stats)
                      ^
/Users/builder/jenkins/ws/openms/PR/bldtst/87227eb0/source/src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp:1235:97: error: reference to local binding 'num_entries' declared in enclosing function 'OpenMS::AccurateMassSearchEngine::exportMzTab_'
      OPENMS_LOG_INFO << "  '" << name << "' : " << adduct_stats_unique[name].size() << " (" << num_entries << ")\n";
                                                                                                ^
/Users/builder/jenkins/ws/openms/PR/bldtst/87227eb0/source/src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp:1233:29: note: 'num_entries' declared here
    for (const auto& [name, num_entries] : adduct_stats)
                            ^

Reply:
probably some incomplete C17 support on that particular clang version

Reply:
if it is not supported, it needs a revert to the old loop without structure binding